packages/vats/src/core/boot-psm.js

```diff
@@ -0,0 +1,223 @@
+ *     anchorAssets: { denom: string }[],
+ * }} vatParameters
+ */
+export const buildRootObject = (vatPowers, vatParameters) => {
+  // @ts-expect-error no TS defs for rickety test scaffolding
+  const log = vatPowers.logger || console.info;
+
+  const {
+    anchorAssets: [{ denom: anchorDenom }], // TODO: handle >1?
+    economicCommitteeAddresses,
+  } = vatParameters;
+
+  const { produce, consume } = makePromiseSpace(log);
+  const { agoricNames, agoricNamesAdmin, spaces } = makeAgoricNamesAccess(
+    log,
+    agoricNamesReserved,
+  );
+  produce.agoricNames.resolve(agoricNames);
+  produce.agoricNamesAdmin.resolve(agoricNamesAdmin);
+
+  const runBootstrapParts = async (vats, devices) => {
+    /** TODO: BootstrapPowers type puzzle */
+    /** @type { any } */
+    const allPowers = harden({
+      vatPowers,
+      vatParameters,
+      vats,
+      devices,
+      produce,
+      consume,
+      ...spaces,
+      // ISSUE: needed? runBehaviors,
+      // These module namespaces might be useful for core eval governance.
+      modules: {
+        utils: { ...utils },
+      },
+    });
+    const manifest = {
+      ...CHAIN_BOOTSTRAP_MANIFEST,
+      ...WALLET_FACTORY_MANIFEST,
+      ...PSM_GOV_INSTALL_MANIFEST,
+      ...ECON_COMMITTEE_MANIFEST,
+      ...PSM_MANIFEST,
+    };
+    const powersFor = name => {
+      const permit = manifest[name];
+      assert(permit, `missing permit for ${name}`);
+      return utils.extractPowers(permit, allPowers);
+    };
+
+    await Promise.all([
+      makeVatsFromBundles(powersFor('makeVatsFromBundles')),
+      buildZoe(powersFor('buildZoe')),
+      makeBoard(powersFor('makeBoard')),
+      makeBridgeManager(powersFor('makeBridgeManager')),
+      makeChainStorage(powersFor('makeChainStorage')),
+      makeAddressNameHubs(powersFor('makeAddressNameHubs')),
+      publishAgoricNames(powersFor('publishAgoricNames'), {
+        options: {
+          agoricNamesOptions: { topLevel: Object.keys(agoricNamesReserved) },
+        },
+      }),
+      startWalletFactory(powersFor('startWalletFactory')),
+      mintInitialSupply(powersFor('mintInitialSupply')),
+      addBankAssets(powersFor('addBankAssets')),
+      startTimerService(powersFor('startTimerService')),
+      // centralSupply, mintHolder, walletFactory
+      installBootContracts(powersFor('installBootContracts')),
+      installGovAndPSMContracts(powersFor('installGovAndPSMContracts')),
+      startEconomicCommittee(powersFor('startEconomicCommittee'), {
+        options: {
+          econCommitteeOptions: {
+            committeeSize: economicCommitteeAddresses.length,
+          },
+        },
+      }),
+      makeAnchorAsset(powersFor('makeAnchorAsset'), {
+        options: { anchorOptions: { denom: anchorDenom } },
+      }),
+      startPSM(powersFor('startPSM'), {
+        options: { anchorOptions: { denom: anchorDenom } },
+      }),
+      // Finally, allow bootstrap powers to be granted
```

Comment:
why "Finally"? don't all these execute in parallel?

Reply:
Yes, they do. Perhaps "Additionally" would be better? Perhaps I should have documented the part where this list contains *only* the things necessary to run the PSM and to do smart wallet transactions.